FinSearchUnified/Bundle/SearchBundleFindologic/ResponseParser/Xml21ResponseParser.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic\ResponseParser;
+
+use Exception;
+use FINDOLOGIC\Api\Responses\Xml21\Properties\LandingPage;
+use FINDOLOGIC\Api\Responses\Xml21\Properties\Promotion as ApiPromotion;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ResponseParser\QueryInfoMessage\CategoryInfoMessage;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ResponseParser\QueryInfoMessage\QueryInfoMessage;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ResponseParser\QueryInfoMessage\SearchTermQueryInfoMessage;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ResponseParser\QueryInfoMessage\VendorInfoMessage;
+use Shopware\Bundle\StoreFrontBundle\Service\ProductNumberServiceInterface;
+
+class Xml21ResponseParser extends ResponseParser
+{
+    /**
+     * @return string[]
+     */
+    public function getProducts()
+    {
+        $foundProducts = [];
+
+        try {
+            $container = Shopware()->Container();
+            /** @var ProductNumberServiceInterface $productService */
+            $productService = $container->get('shopware_storefront.product_number_service');
+
+            foreach ($this->response->getProducts() as $product) {
+                try {
+                    $articleId = $product->getId();
+
+                    $ordernumber = $productService->getMainProductNumberById($articleId);
+
+                    if ($articleId === '' || $articleId === null) {
+                        continue;
+                    }
+                    $baseArticle = [];
+                    $baseArticle['orderNumber'] = $ordernumber;
+                    $baseArticle['detailId'] = $this->getDetailIdForOrdernumber($ordernumber);
+                    $foundProducts[$articleId] = $baseArticle;
+                } catch (Exception $ex) {
+                    // No Mapping for Search Results
+                    continue;
+                }
+            }
+        } catch (Exception $ex) {
+            // Logging Function
+        }
+
+        return $foundProducts;
+    }
+
+    /**
+     * @param $ordernumber
+     *
+     * @return mixed
```

Comment:
`mixed` is not really expressive and tells you that you do not know what is being returned there, which is not true. Based on the implementation this looks like it returns `string|bool`.

Please add a doc for `$db` that shows that it is of type `Enlight_Components_Db_Adapter_Pdo_Mysql`.